Corgibytes.Freshli.Lib.Test/Integration/Languages/Ruby/RubyGemsRepositoryTest.cs

```diff
@@ -1,100 +1,87 @@
 using System;
+using System.Collections.Generic;
 using Corgibytes.Freshli.Lib.Languages.Ruby;
 using Xunit;
 
 namespace Corgibytes.Freshli.Lib.Test.Integration.Languages.Ruby
 {
-    public class RubyGemsRepositoryTest
+    public class RubyGemsRepositoryTest : RepositoryTestFixture<RubyGemsRepositoryTest>
     {
+        public override IPackageRepository Repository => new RubyGemsRepository();
 
-        [Fact]
-        public void VersionInfoCorrectlyCreatesVersion()
+        public override TheoryData<IList<string>, IList<int>, string> DataForTestingVersionInfo => new()
         {
-            var repository = new RubyGemsRepository();
-            var versionInfo = repository.VersionInfo("tzinfo", "1.2.7");
-            var expectedDate = new DateTime(2020, 04, 02);
-
-            Assert.Equal("1.2.7", versionInfo.Version);
-            Assert.Equal(expectedDate, versionInfo.DatePublished);
-        }
-
-        [Fact]
-        public void VersionInfoCorrectlyCreatesPreReleaseVersion()
+            {
+                new[] { "tzinfo", "1.2.7" },
+                new[] { 2020, 04, 02, 21, 42, 11 },
+                "1.2.7"
+            },
+            {
+                new[] { "git", "1.6.0.pre1" },
+                new[] { 2020, 01, 20, 20, 50, 43 },
+                "1.6.0.pre1"
+            }
+        };
+
+        public override TheoryData<IList<object>, string, IList<int>> DataForTestingLatestWithOptionalPreRelease => new()
         {
-            var repository = new RubyGemsRepository();
-            var versionInfo = repository.VersionInfo("git", "1.6.0.pre1");
-            var expectedDate = new DateTime(2020, 01, 20);
-
-            Assert.Equal("1.6.0.pre1", versionInfo.Version);
-            Assert.Equal(expectedDate, versionInfo.DatePublished);
-        }
-
-        [Fact]
-        public void LatestAsOfCorrectlyFindsLatestVersion()
+            {
+                new object[] { "git", new[] { 2020, 02, 01, 00, 00, 00 }, false },
+                "1.5.0",
+                new[] { 2018, 08, 10, 07, 58, 25 }
+            },
+            {
+                new object[] { "git", new[] { 2020, 02, 01, 00, 00, 00 }, true },
+                "1.6.0.pre1",
+                new[] { 2020, 01, 20, 20, 50, 43 }
+            }
+        };
+
+        [Theory(Skip = "Skipping")]
+        [InlineData(null, null, null)]
+        public override void LatestWithMatchExpression(
+          object[] _methodParams,
+          string _expectedVersion,
+          int[] _expectedDateParts
```

Comment:
I kind of like that the warning messages show up here. Gives me an extra incentive to address them.